packages/valory/skills/trader_abci/handlers.py

```diff
@@ -172,6 +204,170 @@ def _send_ok_response(
+
+        if not user_prompt:
+            self._handle_bad_request(http_msg, http_dialogue)
+
+        # Format the prompt
+        prompt_template = CHATUI_PROMPT.format(
+            user_prompt=user_prompt,
+            current_trading_strategy=self._load_chatui_param("trading_strategy"),
+        )
+
+        # Prepare payload data
+        payload_data = {
+            "prompt": prompt_template,
+            "schema": build_chatui_llm_response_schema(),
+        }
+
+        self.context.logger.info(f"Payload data: {payload_data}")
+
+        # Create LLM request
+        srr_dialogues = cast(SrrDialogues, self.context.srr_dialogues)
+        request_srr_message, srr_dialogue = srr_dialogues.create(
+            counterparty=str(GENAI_CONNECTION_PUBLIC_ID),
+            performative=SrrMessage.Performative.REQUEST,
+            payload=json.dumps(payload_data),
+        )
+
+        # Prepare callback args
+        callback_kwargs = {"http_msg": http_msg, "http_dialogue": http_dialogue}
+        self._send_message(
+            request_srr_message,
+            srr_dialogue,
+            self._handle_chatui_llm_response,
+            callback_kwargs,
+        )
+
+    def _handle_chatui_llm_response(
+        self,
+        llm_response_message: SrrMessage,
+        dialogue: Dialogue,  # pylint: disable=unused-argument
+        http_msg: HttpMessage,
+        http_dialogue: HttpDialogue,
+    ) -> None:
+        """
+        Handle the response from the LLM.
+
+        :param llm_response_message: the SrrMessage with the LLM output
+        :param dialogue: the Dialogue
+        :param http_msg: the original HttpMessage
+        :param http_dialogue: the original HttpDialogue
+        """
+        self.context.logger.info(
+            f"LLM response payload: {llm_response_message.payload}"
+        )
+
+        llm_response = json.loads(llm_response_message.payload).get("response", "{}")
+        llm_response_json = json.loads(llm_response)
+
+        llm_message = llm_response_json.get("message", "")
+        updated_agent_config = llm_response_json.get("updated_agent_config", {})
+
+        updated_params = {}
+        if updated_agent_config:
+            updated_trading_strategy: str = updated_agent_config.get(
+                "trading_strategy", None
+            )
+            if updated_trading_strategy:
+                if updated_trading_strategy not in AVAILABLE_TRADING_STRATEGIES:
+                    self.context.logger.error(
+                        f"Unsupported trading strategy: {updated_trading_strategy}"
+                    )
+                else:
+                    # Store the trading strategy in the chatui param store
+                    self._store_chatui_param(
+                        "trading_strategy", updated_trading_strategy
+                    )
+                    updated_params.update(
+                        {"trading_strategy": updated_trading_strategy}
+                    )
```

Comment:
Use early return instead to avoid too much nesting and make the code more readable.